api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -288,4 +288,15 @@ message GrpcJsonTranscoder {
   //
   // If unset, the current stream buffer size is used.
   google.protobuf.UInt32Value max_response_body_size = 16 [(validate.rules).uint32 = {gt: 0}];
+
+  // If true, the transcoder will honor route naming conventions for the below
+  // suffix applied to the route.
+  //
+  // "_skip_transcoding": the transcoder will skip transcoding for the route.
+  //
+  // This is a workaround when per_filter_config isn't available for the route
+  // config. For example, istio control plane.
+  // TODO: remove this feature after istio supports per filter config in their
+  // first-class route APIs.
+  bool use_route_naming_conventions = 17;
```

Comment:
Can we come up with something better? I don't think Istio treats xDS names are stable (and likely change with infra changes).

Does EnvoyFilter not have better filtering condition?

Reply:
We thought about using EnvoyFilter to patch the route component. But was told not to do so because it's also not stable. Do you have some ideas?

Reply:
It's not stable, but so are the rest of Istio APIs to be replaced by Gateway long term. If your usecase (ESF?) can be covered by https://istio.io/latest/blog/2022/getting-started-gtwapi/, I think we need to start making extensions be configured that way. At the very least, Istio could add some annotations on xDS produced by Gateway API per route.

cc @costinm

Reply:
>Istio could add some annotations on xDS produced by Gateway API per route

That would be great. Thanks.

We haven't on boarded Gateway API yet, is there any option to unblock us for the short-term?

@ygong

Reply:
Definitely -1 to having this in Envoy. I had previously mentioned this may be an acceptable short term solution when I thought you were talking about writing your own custom filter with vendor-specific logic that could be coupled to Istio's naming; making Envoy/XDS core do this seems inappropriate.

Reply:
Thank you both. Yes, unfortunately we have to customize this filter. Looks like there's no way for istio to configure per route behavior of this filter?

Reply:
Seems like our best option forward is to use `EnvoyFilter`, what are the concerns about stability for it? I feel it is not a huge concern if we have the proper integration testing in place?